exporter/containerimage/writer.go

```diff
@@ -214,10 +239,74 @@ func (ic *ImageWriter) exportLayers(ctx context.Context, refCfg cacheconfig.RefC
 	return out, err
 }
 
+func (ic *ImageWriter) exportAttestations(ctx context.Context, s session.Group, desc *ocispecs.Descriptor, attestations ...exporter.Attestation) ([]intoto.Statement, error) {
+	span, ctx := tracing.StartSpan(ctx, "export attestations")
+
+	eg, ctx := errgroup.WithContext(ctx)
+	attestationsDone := oneOffProgress(ctx, "exporting attestations")
+
+	statements := make([]intoto.Statement, len(attestations))
+
+	for i, att := range attestations {
+		switch att := att.(type) {
+		case *exporter.InTotoAttestation:
+			mount, err := att.PredicateRef.Mount(ctx, true, s)
+			if err != nil {
+				return nil, err
+			}
+
+			lm := snapshot.LocalMounter(mount)
+			src, err := lm.Mount()
+			if err != nil {
+				return nil, err
+			}
+			defer lm.Unmount()
+
+			rawPredicate, err := os.ReadFile(path.Join(src, att.PredicatePath))
+			if err != nil {
+				return nil, err
+			}
+			var predicate interface{}
```

Comment:
This should be `json.RawMessage`. Otherwise we are converting the types, messing with order and possibly other bad things.

We could try to parse it to make sure it is a valid JSON in here though.